compiler-rt/lib/asan/asan_globals.cpp

```diff
@@ -80,14 +80,16 @@ static bool IsAddressNearGlobal(uptr addr, const __asan_global &g) {
 }
 
 static void ReportGlobal(const Global &g, const char *prefix) {
+  DataInfo info;
+  bool symbolized = Symbolizer::GetOrInit()->SymbolizeData(g.beg, &info);
   Report(
-      "%s Global[%p]: beg=%p size=%zu/%zu name=%s module=%s dyn_init=%zu "
+      "%s Global[%p]: beg=%p size=%zu/%zu name=%s source=%s module=%s "
+      "dyn_init=%zu "
       "odr_indicator=%p\n",
       prefix, (void *)&g, (void *)g.beg, g.size, g.size_with_redzone, g.name,
-      g.module_name, g.has_dynamic_init, (void *)g.odr_indicator);
+      g.module_name, info.module, g.has_dynamic_init, (void *)g.odr_indicator);
```

Comment:
`info.module` might not be valid if `symbolized` is false?

Reply:
Good catch. It's `nullptr` and will be printed as `<null>`. `(symbolized ? info.module : "")` will be better.